compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt

```diff
@@ -370,67 +376,98 @@ internal abstract class ComposeBridge(
     }
 
     private fun refreshWindowFocus() {
-        platform.windowInfo.isWindowFocused = window?.isFocused ?: false
+        windowInfo.isWindowFocused = window?.isFocused ?: false
         keyboardModifiersRequireUpdate = true
     }
 
-    protected inner class DesktopPlatform : Platform by Platform.Empty {
-        override fun setPointerIcon(pointerIcon: PointerIcon) {
-            component.cursor =
-                (pointerIcon as? AwtCursor)?.cursor ?: Cursor(Cursor.DEFAULT_CURSOR)
-        }
+    private inner class DesktopViewConfiguration : ViewConfiguration by EmptyViewConfiguration {
+        override val touchSlop: Float get() = with(platformComponent.density) { 18.dp.toPx() }
+    }
 
-        override fun accessibilityController(owner: SemanticsOwner) =
-            AccessibilityControllerImpl(owner, platformComponent, onFocusReceived = {
-                requestNativeFocusOnAccessible(it)
-            })
+    private inner class DesktopFocusManager : FocusManager {
+        override fun clearFocus(force: Boolean) {
+            val root = component.rootPane
+            root?.focusTraversalPolicy?.getDefaultComponent(root)?.requestFocusInWindow()
+        }
 
-        override val windowInfo = WindowInfoImpl()
+        override fun moveFocus(focusDirection: FocusDirection): Boolean =
+            when (focusDirection) {
+                FocusDirection.Next -> {
+                    val toFocus = component.focusCycleRootAncestor?.let { root ->
+                        val policy = root.focusTraversalPolicy
+                        policy.getComponentAfter(root, component)
+                            ?: policy.getDefaultComponent(root)
+                    }
+                    val hasFocus = toFocus?.hasFocus() == true
+                    !hasFocus && toFocus?.requestFocusInWindow(FocusEvent.Cause.TRAVERSAL_FORWARD) == true
+                }
 
-        override val textInputService = PlatformInput(platformComponent)
+                FocusDirection.Previous -> {
+                    val toFocus = component.focusCycleRootAncestor?.let { root ->
+                        val policy = root.focusTraversalPolicy
+                        policy.getComponentBefore(root, component)
+                            ?: policy.getDefaultComponent(root)
+                    }
+                    val hasFocus = toFocus?.hasFocus() == true
+                    !hasFocus && toFocus?.requestFocusInWindow(FocusEvent.Cause.TRAVERSAL_BACKWARD) == true
+                }
```

Comment:
Was this copy/pasted from somewhere else? Can we simplify it? Maybe
```
        override fun moveFocus(focusDirection: FocusDirection): Boolean {
            val (componentGetter, cause) = when(focusDirection) {
                FocusDirection.Next ->
                    FocusTraversalPolicy::getComponentAfter to FocusEvent.Cause.TRAVERSAL_FORWARD
                FocusDirection.Previous ->
                    FocusTraversalPolicy::getComponentAfter to FocusEvent.Cause.TRAVERSAL_BACKWARD
                else -> return false
            }

            val toFocus = component.focusCycleRootAncestor?.let { root ->
                val policy = root.focusTraversalPolicy
                componentGetter(policy, root, component)
                    ?: policy.getDefaultComponent(root)
            } ?: return false


            return !toFocus.hasFocus() && toFocus.requestFocusInWindow(cause)
        }
```

Reply:
> Was this copy/pasted from somewhere else?

Yes, it's just moved from another line without change.